src/app/shared/register-github-app/register-github-app.component.ts

```diff
@@ -2,16 +2,37 @@ import { Component, Input } from '@angular/core';
 import { Dockstore } from '../dockstore.model';
 import { SessionQuery } from '../session/session.query';
 import { UserQuery } from '../user/user.query';
+import { Router } from '@angular/router';
+import { MatDialogRef } from '@angular/material/dialog';
+import { RegisterGithubAppModalComponent } from '../../workflow/register-workflow-modal/register-github-app-modal/register-github-app-modal.component';
 
 @Component({
   selector: 'app-register-github-app',
   templateUrl: './register-github-app.component.html',
 })
 export class RegisterGithubAppComponent {
   public Dockstore = Dockstore;
-  public gitHubAppInstallationLink$ = this.sessionQuery.gitHubAppInstallationLink$;
+  public gitHubAppInstallationLink$ = this.sessionQuery.gitHubAppInstallationLandingPageLink$;
   public isUsernameChangeRequired$ = this.userQuery.isUsernameChangeRequired$;
   @Input() public entryType: string;
 
-  constructor(private sessionQuery: SessionQuery, private userQuery: UserQuery) {}
+  constructor(
+    private sessionQuery: SessionQuery,
+    private userQuery: UserQuery,
+    private router: Router,
+    public parentDialogRef: MatDialogRef<RegisterGithubAppModalComponent>
+  ) {}
+
+  public registerGitHubApp(observable) {
```

Comment:
If I understand correctly, a new tab opens for the GitHub installation and the current tab redirects to the landing page. What happens to the new tab when the user saves their configuration? The `state` query parameter of the GH app installation link is `/github-landing-page`, which means that the GitHub callback would redirect the new tab to the GitHub landing page once the user clicks save...unless this new tab is automatically closed?

Pondering based on my assumptions, I wonder if it's best to leave the current tab as is (don't redirect), open the installation in a new tab, and that new tab gets redirected to the landing page. That way, the user still has their previous page open in their first tab, which addresses one of my other comments in this PR about the user not being able to get back to My <Entry>.